src/stores/search.js

```diff
@@ -0,0 +1,172 @@
+   *
+   * @returns {import('@nuxtjs/composition-api').ComputedRef<import('../store/types').ApiQueryParams>}
+   */
+  const searchQueryParams = computed(() => {
+    // Ensure that q filter always comes first
+    return Object.keys(state.query).reduce(
+      (obj, key) => {
+        return !['q'].includes(key) && state.query[key].length
+          ? { ...obj, [key]: state.query[key] }
+          : obj
+      },
+      { q: state.query.q.trim() }
+    )
+  })
+
+  /**
+   * Called when `q` search term or `searchType` are changed. Updates the filters for
+   * the current search type, and then updates the search query accordingly.
+   * @param {{ q?: string, searchType?: import('../store/types').SupportedSearchType }} params
+   */
+  function updateQuery(params = {}) {
+    const { q, searchType } = params
+    if (q) {
+      state.query.q = q.trim()
+    }
+    if (searchType && searchType !== state.searchType) {
+      setSearchType(searchType)
+      filterStore.clearOtherMediaTypeFilters({ searchType })
+    }
+    updateQueryFromFilters()
+  }
+
+  /**
+   * Called when a /search path is server-rendered.
+   * @param {{ path: string, query: { [key: string]: string} }} params
+   */
+  function setSearchStateFromUrl({ path, query }) {
+    if (query.q) {
+      state.query.q = query.q.trim()
+    }
+
+    const searchType = queryStringToSearchType(path)
+    setSearchType(searchType)
+
+    filterStore.updateFiltersFromUrl(query, searchType)
+    updateQueryFromFilters()
+  }
+  /**
+   * After a change in filters, updates the query. If the filter is not selected, query should be set to blank string "".
+   * @param {object} [params]
+   * @param {import('../store/types').SupportedSearchType} [params.mediaType]
+   * @param {string} [params.q]
+   */
+  function updateQueryFromFilters(params = {}) {
+    const queryFromFilters = filtersToQueryData(
+      filterStore.filters,
+      params.mediaType || state.searchType,
+      true
+    )
+
+    // If the filter was unchecked, its value in `queryFromFilters` would be falsy, ''.
+    // So we check if the key exists, not if the value is not falsy.
+    const changedKeys = /** @type {import('../store/types').QueryKey[]} */ (
+      Object.keys(queryFromFilters)
+    )
+    const queryKeys = /** @type {import('../store/types').QueryKey[]} */ (
+      Object.keys(state.query)
+    )
+    state.query = queryKeys.reduce((obj, key) => {
+      if (key === 'q') {
+        obj[key] = params?.q || state.query.q
+      } else {
+        obj[key] = changedKeys.includes(key) ? queryFromFilters[key] : ''
+      }
+      return obj
+    }, /** @type {import('../store/types').Query} */ ({}))
+  }
+
+  /**
+   * Toggles a filter's checked parameter
+   * @param {{ filterType: import('../store/types').FilterCategory, codeIdx?: number, code?: string}} params
+   */
+  function toggleFilter(params) {
+    filterStore.toggleFilter(params)
+    updateQueryFromFilters()
+  }
```

Comment:
Same deal here as the search type. Should we just have a watcher to update the search store whenever the filter store sees a change? It would eliminate the confusing duplicated API.

Reply:
I tried using pinia $subscribe to replace the calls for update:
`filterStore.$subscribe(() => updateQueryFromFilters())` inside the searchStore, but it did not call the `update` function on the server, and did not update the query (and the path) ...
Do you think opening a separate issue and solving it in a different PR would be a good idea?